lib/mint/web_socket.ex

```diff
@@ -112,7 +112,6 @@ defmodule Mint.WebSocket do
 
   alias __MODULE__.{Utils, Extension, Frame}
   alias Mint.{WebSocketError, WebSocket.UpgradeFailureError}
-  alias Mint.{HTTP1, HTTP2}
```

Comment:
Does it cause Dialyzer problems to use these modules? I would prefer to use these directly where they were since it indicates that we know what the protocol is.

Reply:
Dialyzer complains when it sees a `Mint.HTTP.t()` being passed to either the `HTTP1` or `HTTP2` modules. It also complains when it sees a `HTTP1.t()` being returned where it expects a `Mint.HTTP.t()`

It's possible to convince Dialyzer through the use of some functions with explicit specs, since Dialyzer doesn't seem to look very carefully at opaque types.

```elixir
@spec convince_dialyzer(:http1, Mint.HTTP.t()) :: HTTP1.t()
def convince_dialyzer(:http1, conn), do: conn

@spec convince_dialyzer(:http2, Mint.HTTP.t()) :: HTTP2.t()
def convince_dialyzer(:http2, conn), do: conn

@spec convince_dialyzer(HTTP1.t() | HTTP2.t()) :: Mint.HTTP.t()
def convince_dialyzer(conn), do: conn
```

I'm not a big fan of this approach, though.

Reply:
Yeah, switching to `Mint.HTTP` is ok then 👍